src/wallet/wallet.h

```diff
@@ -107,6 +107,7 @@ static const bool DEFAULT_WALLET_RBF = true;
 static const bool DEFAULT_WALLETBROADCAST = true;
 static const bool DEFAULT_DISABLE_WALLET = false;
 static const bool DEFAULT_WALLETCROSSCHAIN = false;
+static const std::string DEFAULT_WALLET_LABEL = "";
```

Comment:
Conceptually there isn't a default label, the record label is just empty.

What if you return from `LabelFromValue` the empty string instead? It will also remove the need for all the `.value_or(DEFAULT_WALLET_LABEL);`.

Reply:
Some RPCs require the label to be optional for filtering. `getaddressesbylabel` will return all addresses if the label is nullopt for example.
I could hard-code the `""` in `.value_or` calls if you prefer though.

Reply:
Maybe you are referring to `listsinceblock`? (as far as I can see, it's the only place where the label is used as optional), `getaddressesbylabel` uses a string not an optional.

Check this out: https://github.com/furszy/bitcoin-core/commit/8ce2d8965537a889fe88ca1a891763b35aba32ad. (If you like it, feel free to squash it on the first commit).

Reply:
The CI doesn't pass on your commit because of `listsinceblock`. A label can be empty in this RPC.

https://cirrus-ci.com/task/6056876765347840?logs=ci#L3626

Reply:
will check it tonight 👍🏼. I coded a quick example. Shouldn't be hard to correct that single case.

Reply:
Ready, fix was simple: https://github.com/furszy/bitcoin-core/commit/f8da21cb6eec0d62b2d3ed766201aca182b47342.